### ISSUE TITLE
False positive for ComposeModifierReused

### ISSUE THREAD

**Issue by ansman** [slackhq/compose-lints#291]
Here is an example:
```kotlin
@Composable
fun Example(
    modifier: Modifier = Modifier,
    content: @Composable BoxScope.() -> Unit,
) {
    val context = LocalContext.current
    val currentContent by rememberUpdatedState(content)
    val currentModifier by rememberUpdatedState(modifier)
    val overlayView = remember {
        ComposeView(context).apply {
            setContent {
                Box(currentModifier.fillMaxSize(), content = currentContent)
            }
        }
    }
    DisposableEffect(overlayView) {
        onDispose {}
    }
}
```

I'm seeing a lint error on both the `Box` as well as the `DisposableEffect` saying that composable modifiers shouldn't be reused. Commenting out either removes the error.

**Comment by ZacSweers** [comment]
That's strange. Does that reproduce in a test?
